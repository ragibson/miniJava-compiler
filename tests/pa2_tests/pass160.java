// PA1 parse refs pass
class Test {

    void p() {
        A a = 23;
	boolean b = c;
    }
}

