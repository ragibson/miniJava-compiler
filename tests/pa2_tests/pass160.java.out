********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "Test" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [2]
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     ClassType
  .   .   .       "A" Identifier
  .   .   .     "a" varname
  .   .   .   LiteralExpr
  .   .   .     "23" IntLiteral
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     BOOLEAN BaseType
  .   .   .     "b" varname
  .   .   .   RefExpr
  .   .   .     IdRef
  .   .   .       "c" Identifier
=============================================
